extensions/2.0/Khronos/KHR_materials_volume/README.md

```diff
@@ -0,0 +1,268 @@
+</figure>
+
+The volume extension has to be combined with `KHR_materials_transmission` or `KHR_materials_translucency`. Light that falls onto the volume boundary may enter the volume, depending on the transmission or translucency of the surface's BSDF. Inside the volume, light might be absorbed or scattered by particles in the medium, and eventually hit some surface from inside the volume. It can be a different surface and will most probably be a different point in space compared to the entrance point. At this exit point, if the material properties allow, light may leave the volume.
+
+<figure style="text-align:center">
+<img src="./figures/volume.svg"/>
+<figcaption><em>Interaction of light rays inside the volume and at the boundaries. The volume is homogeneous and has an index of refraction of 1.5.</figcaption>
+</em></figure>
+
+When light interacts with the surface, it is reflected or refracted at microfacets, taking the roughness of the material into account. The index of refraction is taken from `KHR_materials_ior`.
+
+## Extending Materials
+
+The volumetric properties are defined by adding the `KHR_materials_volume` extension to any glTF material. A non-zero thickness switches from thin-walled to volumetric behavior. This requires a manifold/closed mesh. The properties of the medium are not texturable, it is assumed to be homogeneous.
+
+```json
+materials: [
+    {
+        "extensions": {
+            "KHR_materials_volume": {
+                "thicknessFactor": 1.0,
+                "attenuationDistance":  0.006,
+                "attenuationColor": [ 0.5, 0.5, 0.5 ],
+                "subsurfaceColor": [ 0.572, 0.227, 0.075 ]
+            }
+        }
+    }
+]
+```
+
+## Properties
+
+The extension defines the following parameters to describe the volume.
+
+| | Type | Description | Required |
+|-|------|-------------|----------|
+| **thicknessFactor** | `number` | The thickness of the volume beneath the surface in meters in the local coordinate system of the node. If the value is 0 the material is thin-walled. Otherwise the material is a volume boundary. The `doubleSided` property has no effect on volume boundaries. | No, default: `0`. |
+| **thicknessTexture** | `textureInfo` | A texture that defines the thickness, stored in the G channel. This will be multiplied by `thicknessFactor`. | No |
+| **attenuationDistance** | `number` | Density of the medium given as the average distance in meters that light travels in the medium before interacting with a particle. | No, default: +Infinity |
```

Comment:
Is this the average distance between particles in the medium that will affect how many times the light will either scatter or be absorbed?

Reply:
Maybe not exactly how many times (the light bounces infinitely), but more the average distance light travels in the medium until it hits a particle. The particle will then either scatter the light or absorb it. How much of it is absorbed or scattered is given by the absorption coefficient (not to be confused with the attenuation coefficient!) and the particle's albedo (i.e., the single scattering color derived from the subsurface color). 

If subsurface color is 0, the single scattering color becomes 0. Thus, there is no scattering, and only in this case (!) absorption coefficient = attenuation coefficient.

If subsurface color is > 0, single scattering color also becomes > 0. Now the particles scatter and absorb the light, according to some ratio.

Note that by parameterizing with attenuation coefficient and particle albedo we cannot construct the case of no absorption, scattering only.

The attenuation distance is the scalar part of the attenuation coefficient. The attenuation coefficient is the inverse of the mean free path length.

attenuation_coefficient = 1 / mean_free_path = -log(attenuation_color) / attenuation_distance